httpx/_content_streams.py

```diff
@@ -331,7 +331,7 @@ def encode(
         else:
             return ByteStream(body=b"")
     elif isinstance(data, dict):
```

Comment:
Seems like we should also check for empty data here, so as to not send an empty `URLEncodedStream`, but instead an empty `ByteStream`:

```suggestion
    elif isinstance(data, dict) and data:
```

This would put us in line with what requests does:

```python
>>> import requests
>>> url = 'https://httpbin.org/post'
>>> r = requests.post(url, data={}, files={})
>>> r.request.headers
{'User-Agent': 'python-requests/2.23.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'Content-Length': '0'}
```

Reply:
Implemented this idea in different line